### ISSUE TITLE
Better post-quantization performance

### ISSUE THREAD

**Issue by pseudotensor** [h2oai/h2ogpt#222]
Notes about Intriguing Properties of Quantization at Scale https://arxiv.org/pdf/2305.19268.pdf

So basically should do:
* bf16 instead of fp16 training, less sensitive to quantization later
* weight decay order 0.05, not too strong at 0.1 and not too weak at 0.01
* layernorm at fp32
* dropout order 0-0.1, not 0.4-0.8
* gradient clipping order 1, not none

Related to 4-bit quant qlora paper: https://huggingface.co/blog/4bit-transformers-bitsandbytes#advanced-usage

**Comment by pseudotensor** [comment]
won't do in this repo.
